### PR TITLE
test: implement test for extract_prefix_and_root

### LINKED ISSUE #105: Add tests for extract_prefix_and_root
## Description 📹

**Add tests for the `extract_prefix_and_root` function located in `src/utils.rs`.**  
Ensure the function's behavior is thoroughly validated under various scenarios, including edge cases.

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   
```bash
   git checkout -b fix-[issue-number]
```

2. **Implement Changes**:  
   Add unit tests for the `extract_prefix_and_root` function. Place the tests in the appropriate test module within `src/utils.rs` or a dedicated test file.  
   
   **Test cases to include**:
   - [ ] Valid inputs with distinct prefix and root combinations.
   - [ ] Inputs with no prefix or root to test edge cases.
   - [ ] Empty string input.
   - [ ] Unexpected formats or invalid data.
   - [ ] Any other cases relevant to the function's specification.

   Example (to be expanded upon):

   ```rust
   #[cfg(test)]
   mod tests {
       use super::*;

       #[test]
       fn test_extract_prefix_and_root_valid_input() {
           let input = ...;
           let (prefix, root) = extract_prefix_and_root(input).unwrap();
           assert_eq!(prefix, ...;
           assert_eq!(root, ...);
       }

       // Add more cases here
   }
   ```

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   
```bash
   git commit -m "Test: Add unit tests for extract_prefix_and_root function"
```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing Close #[issue_id].
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by mayasimi**
Closes #105 

testpass

Ready for review

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -166,3 +166,76 @@ pub async fn fetch_image_url(config: &Config, url: &str) -> String {
         Err(_) => "Error fetching data".to_string(),
     }
 }
+
+
+#[cfg(test)]
```
Comment (Marchand-Nicolas) [inline src/utils.rs]: Please put it in src/tests/utils.rs (you'll have to create the folders)

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

**Comment by mayasimi** [comment]
@Marchand-Nicolas 

implemented the separation of test in a different file. ready for review

test passed:

Screenshot from 2024-11-24 15-03-38

**Comment by mayasimi** [comment]
@Marchand-Nicolas ready for review

File: src/utils.rs [hunk 2]
```diff
@@ -166,3 +166,4 @@ pub async fn fetch_image_url(config: &Config, url: &str) -> String {
         Err(_) => "Error fetching data".to_string(),
     }
 }
+git
```
Comment (Marchand-Nicolas) [inline src/utils.rs]: I guess you forgot to remove that

**Review by Marchand-Nicolas — APPROVED** [review]
I made a little fix, now lgtm! Well done it's working perfectly :)
